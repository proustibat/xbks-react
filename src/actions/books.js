// SET_BOOKS
export const setBooks = books => ( {
    type: 'SET_BOOKS',
    books
} );
