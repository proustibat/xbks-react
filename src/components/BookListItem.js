import React from 'react';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const BookListItem = ( { isbn, title, price, synopsis } ) => (
    <Link to={ `/book/${ isbn }` } className="list-item">

        <div>
            <h3 className="list-item__title">{ title }</h3>
            <span className="list-item__sub-title">{ synopsis[ 0 ] }</span>
        </div>

        <h3 className="list-item__data">{ numeral( price ).format( '0.00$' ) }</h3>

    </Link>
);

export default BookListItem;
