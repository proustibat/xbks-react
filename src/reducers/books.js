const booksReducerDefaultState = [];
export default ( state = booksReducerDefaultState, action ) => {
    switch ( action.type ) {
        case 'SET_BOOKS':
            return action.books;
        default:
            return state;
    }
};
