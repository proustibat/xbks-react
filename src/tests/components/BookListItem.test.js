import React from 'react';
import { shallow } from 'enzyme';
import BookListItem from '../../components/BookListItem';
import books from "../fixtures/books";

test( 'Should render BookListItem component correctly', () => {
    const bookData = books[ 0 ];
    const wrapper = shallow( <BookListItem
        { ...bookData }
    /> );
    expect( wrapper ).toMatchSnapshot();
} );
