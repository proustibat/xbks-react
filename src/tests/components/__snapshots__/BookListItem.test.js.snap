// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render BookListItem component correctly 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/book/c8fabf68-8374-48fe-a7ea-a00ccd07afff"
>
  <span>
    <img
      alt=""
      className="list-item__preview"
      src="http://henri-potier.xebia.fr/hp0.jpg"
    />
  </span>
  <h3
    className="list-item__title"
  >
    Henri Potier à l'école des sorciers
  </h3>
  <span
    className="list-item__data"
  >
    35.00$
  </span>
</Link>
`;
